### PR TITLE
test: add product attributes test

### PR DESCRIPTION
**Pull request by doproiescu-plenty**
## Why:

Closes: #ID

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    it('updateValue updates attribute values', () => {
+        const { updateValue, attributeValues } = useProductAttributes();
+        updateValue(1, 1);
+
+        expect(attributeValues.value).toEqual({});
+
+        updateValue(1, 2);
+
+        expect(attributeValues.value).toEqual({});
```
Comment (ksted) [inline apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts]: **todo**: Split into multiple test cases. Every test case should assert against a single outcome, not multiple.

**question**: Why do you always assert against an empty object? I don't understand what we're validating here if the expected value never changes.

File: apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    });
+
+    it('getCombination returns the correct combination', () => {
+        const { getCombination } = useProductAttributes();
+        const combination = getCombination();
+
+        expect(combination).toEqual(null);
+    });
```
Comment (ksted) [inline apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts]: **todo**: Add a test case for an outcome other than `null`.

File: apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    it('getValue returns the correct attribute value', () => {
+        const { getValue } = useProductAttributes();
+        const value = getValue(1);
+
+        expect(value).toBe(undefined);
```
Comment (ksted) [inline apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts]: **todo**: add a test case for an outcome other than `undefined`.

**Review by ksted — CHANGES_REQUESTED** [review]

File: apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import ProductFixture from "./../../../utils/__tests__/__fixtures__/Product";
+
+describe('useProductAttributes', () => {
+    it('initializes state correctly', () => {
```
Comment (ksted) [inline apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts]:
```suggestion
    it('should initialize state correctly', () => {
```
Always begin test cases with `it('should...')`

File: apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        const { attributes, attributeValues, combinations, itemId, variationId } = useProductAttributes();
+
+        expect(attributes.value).toEqual([]);
+        expect(attributeValues.value).toEqual({});
+        expect(combinations.value).toEqual([]);
+        expect(itemId.value).toBe(0);
+        expect(variationId.value).toBe(0);
+    });
+
+    it('updateValue updates attribute values', () => {
```
Comment (ksted) [inline apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts]:
```suggestion
    it('should update attribute values', () => {
```
Avoid specifying the function name in the test case. The case should just describe the expected outcome without caring about how it's achieved.

File: apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        const { setAttribute, updateValue, attributeValues } = useProductAttributes();
+        setAttribute(ProductFixture, true);
+        updateValue(1, 1);
+
+        expect(attributeValues.value).toEqual({ 1: 1, });
+    });
+
+
+    it('updateValue should return empty object if value is not found', () => {
```
Comment (ksted) [inline apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts]:
```suggestion
    it('should should return empty object if value is not found during update', () => {
```

File: apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        const { setAttribute, updateValue, attributeValues } = useProductAttributes();
+        setAttribute(ProductFixture, true);
+        updateValue(1, 2);
+
+        expect(attributeValues.value).toEqual({});
+    });
+
+    it('getCombination returns the correct combination', () => {
```
Comment (ksted) [inline apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts]:
```suggestion
    it('should get the correct combination of attributes', () => {
```

File: apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+            attributes: [],
+            isSalable: true,
+            unitCombinationId: 9,
+            unitId: 5,
+            unitName: "2 liter",
+            variationId: 1100,
+        });
+    });
+
+    it('getValue returns the correct attribute value', () => {
```
Comment (ksted) [inline apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts]:
```suggestion
    it('should get the correct attribute value', () => {
```

File: apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+        expect(itemId.value).toBe(0);
+        expect(variationId.value).toBe(0);
+    });
+
+    it('updateValue updates attribute values', () => {
+        const { setAttribute, updateValue, attributeValues } = useProductAttributes();
+        setAttribute(ProductFixture, true);
+        updateValue(1, 1);
+
+        expect(attributeValues.value).toEqual({ 1: 1, });
```
Comment (ksted) [inline apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts]:
```suggestion
        expect(attributeValues.value).toEqual({ 1: 1 });
```

File: apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts (hunk 6 above)
Comment (ksted) [reply apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts]: **question**: Just to be sure, does this only refer to the initial state or is this always true? In other words, what should happen if `attributesValues.value` is already set and you try to update with an invalid value? Does it get unset or keep the previous value?

**Review by ksted — CHANGES_REQUESTED** [review]

File: apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts (hunk 6 above)
Comment (doproiescu-plenty) [reply apps/web/composables/useProductAttributes/__tests__/useProductAttributes.spec.ts]: Each time setAttribute is called, state.value.attributeValues = {}. The second parameter of the function, if true, will set the attribute values of the product send in first parameter. So every time the function is called, new attribute values pop will come

**Review by ksted — APPROVED** [review]
